src/Component/Metadata/Resource/Factory/AttributesResourceMetadataCollectionFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Metadata\Resource\Factory;
+
+use Sylius\Component\Resource\Metadata\Operation;
+use Sylius\Component\Resource\Metadata\Operations;
+use Sylius\Component\Resource\Metadata\Resource as ResourceMetadata;
+use Sylius\Component\Resource\Metadata\Resource\ResourceMetadataCollection;
+use Sylius\Component\Resource\Reflection\ClassReflection;
+use Webmozart\Assert\Assert;
+
+final class AttributesResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
+{
+    public function create(string $resourceClass): ResourceMetadataCollection
+    {
+        $resourceMetadataCollection = new ResourceMetadataCollection();
+
+        $attributes = ClassReflection::getClassAttributes($resourceClass);
+
+        foreach ($this->buildResourceOperations($attributes, $resourceClass) as $resource) {
+            $resourceMetadataCollection[] = $resource;
+        }
+
+        return $resourceMetadataCollection;
+    }
+
+    /**
+     * @param \ReflectionAttribute[] $attributes
+     */
+    private function buildResourceOperations(array $attributes, string $resourceClass): array
```

Comment:
```suggestion
    /**
     * @param \ReflectionAttribute[] $attributes
     *
     * @return ResourceMetadata[]
     */
    private function buildResourceOperations(array $attributes, string $resourceClass): array
```
But perhaps the `ResourceMetadata[]` could be even translated to custom array object. WDYT?